gtdynamics/kinematics/KinematicsInterval.cpp

```diff
@@ -44,6 +54,16 @@ NonlinearFactorGraph Kinematics::pointGoalObjectives<Interval>(
   return graph;
 }
 
+template <>
+EqualityConstraints Kinematics::pointGoalConstraints<Interval>(
```

Comment:
Why do we have both objectives and constraints?

Reply:
Sometimes we want to have the point goals as objectives (there are some cases in which not all point goals can be reached), and sometimes we want them to be constraints. I can keep only the constraints if we don't need the objective version.